.github/workflows/release.yml

```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: windows-latest
     strategy:
       matrix:
-        mode: [osx-x64,linux-x64,win-x64]
+        mode: [osx-x64,linux-x64,win-x64,win-arm64,osx-arm64,linux-arm64]
```

Comment:
I think it might be possible to remove some conditional logic later in the file by doing something like this:
```yaml
      matrix:
        mode: [osx-x64,linux-x64,win-x64,win-arm64,osx-arm64,linux-arm64]
        include:
        - mode: osx-x64
          kubectlUrl: 'https://dl.k8s.io/release/v1.27.3/bin/darwin/amd64/kubectl'
          kubectlPath: 'osx\kubectl'
        - mode: linux-x64
          kubectlUrl: 'https://dl.k8s.io/release/v1.27.3/bin/linux/amd64/kubectl'
          kubectlPath: 'linux\kubectl'
        - mode: win-x64
          kubectlUrl: 'https://dl.k8s.io/release/v1.27.3/bin/windows/amd64/kubectl.exe'
          kubectlPath: 'win\kubectl.exe'
        - mode: win-arm64
          kubectlUrl: 'https://dl.k8s.io/release/v1.27.3/bin/windows/arm64/kubectl.exe'
          kubectlPath: 'win\kubectl.exe'
        - mode: osx-arm64
          kubectlUrl: 'https://dl.k8s.io/release/v1.27.3/bin/darwin/arm64/kubectl'
          kubectlPath: 'osx\kubectl'
        - mode: linux-arm64
          kubectlUrl: 'https://dl.k8s.io/release/v1.27.3/bin/linux/arm64/kubectl'
          kubectlPath: 'linux\kubectl'
```
Then later:
```yaml
    - name: Download Kubectl
      uses: Azure/powershell@v1
      with:
        inlineScript: |
          New-Item -Path '${{ github.workspace }}\src\dsc\bin\Release\net7.0\${{ matrix.mode }}\publish\kubectl' -ItemType Directory
          $kubectlPath = '${{ github.workspace }}\src\dsc\bin\Release\net7.0\${{ matrix.mode }}\publish\kubectl\${{ matrix.kubectlPath }}'
          $kubectlDir = Split-Path $kubectlPath
          New-Item -Path $kubectlDir -ItemType Directory
          curl.exe $url -o $kubectlPath
```

I think the same trick could be used for uploading release assets (i.e. we could replace all 6 steps with a single one).

What do you reckon?

Reply:
@peterbom Yep this is good point, we don't use this release.yaml since binaries needs to be signed and GitHub actions doesn't support code signing yet. Pipelines are still internal azure.pipelines (please refer this). I just updated to keep it in sync with current pipelines.

Reply:
updated this.